p521/src/arithmetic/field.rs

```diff
@@ -277,21 +277,9 @@ impl FieldElement {
     /// Returns the square root of self mod p, or `None` if no square root
     /// exists.
     pub fn sqrt(&self) -> CtOption<Self> {
-        // Tonelli-Shank's algorithm for q mod 4 = 3 (i.e. Shank's algorithm)
-        // https://eprint.iacr.org/2012/685.pdf
-        let w = self.pow_vartime(&[
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000000,
-            0x0000000000000080,
-        ]);
-
-        CtOption::new(w, w.square().ct_eq(self))
+        // See explanation @ https://github.com/RustCrypto/elliptic-curves/pull/946#discussion_r1380465035
```

Comment:
Maybe just put the explanation inline rather than linking to a review comment

Reply:
![image](https://github.com/RustCrypto/elliptic-curves/assets/7382032/3c6c1322-cd69-40b1-95de-afb7ec06f567)